app/hooks/use-display-currency.ts

```diff
@@ -249,6 +233,15 @@ export const useDisplayCurrency = () => {
       const { symbol, minorUnitToMajorUnitOffset, showFractionDigits, currencyCode } =
         currencyInfo[moneyAmount.currency]
 
+      if (
+        moneyAmount.currency === DisplayCurrency &&
+        currencyCode !== moneyAmount.currencyCode
+      ) {
+        console.log("currencyCode mismatch", currencyCode, moneyAmount.currencyCode)
+        // todo: we should display the correct currency but this requires `showFractDigits` to come from the backend
+        return NaN.toString()
```

Comment:
I don't think the `NaN.toString()` is the best value to return here. we should say "there is an issue with the display currency, please refresh". will think about a short way to put it.